File: HelloWorldInator.java
public class HelloWorldInator {
    private final VoidFunction helloWorld;
    public HelloWorldInator() {
        helloWorld = () -> {
            System.out.println("Hello, World!");
        };
    }

    public VoidFunction getHelloWorld() {
        return helloWorld;
    }

    public interface VoidFunction {
        void run();
    }
}

File: HelloWorldInator.zig
const std = @import("std");

pub fn main() !void {
    var hello_world_fn = hello_world_inator();
    try hello_world_fn();
}

fn hello_world() !void {
    var stdout_file = try std.io.getStdOut();
    try stdout_file.write("Hello, World!");
}

fn hello_world_inator() @typeOf(hello_world) {
    return hello_world;
}
